scripts/data/lib/widgetXmlParser.ts

```diff
@@ -0,0 +1,41 @@
+import { XMLParser } from "fast-xml-parser";
+import { readFile } from "./util";
+
+type Patterns = { [key: string]: (doc: any) => any }; // boolean indicates if value is required or not
```

Comment:
Can we define the structure instead of generic `any` ?

Reply:
hmm, not really, as in we can't know the structure of the document for sure. So the types would just be pretend-types.
Unless we would use something like zod, which can produce full validation and typing for JSON. But it seems a bit overkill.
I will restrict the types of the extract values a bit though, to get rid of `any` there

Reply:
The problem with `any` is that it suppresses all type checking, so I definitely recommend avoiding it :).

Reply:
the effects on the rest of the application should be mitigated now. The key for that is in the `Values` type..

Reply:
I think this is as good as it can get, given that we're stuck with `any` in so many places anyway :).